src/test/java/fi/oph/akt/service/ContactRequestServiceTest.java

```diff
@@ -120,84 +118,93 @@ public void createContactRequestShouldSaveValidRequest() {
+    final List<EmailData> emailDatas = emailDataCaptor.getAllValues();
 
     assertEquals(3, emailDatas.size()); // 2 + 1 copy to foo@bar
 
+    translators.forEach(t ->
+      assertEquals(
+        1,
+        emailDatas
+          .stream()
+          .filter(e -> e.recipientName().equals(t.getFullName()) && e.recipientAddress().equals(t.getEmail()))
+          .count()
+      )
+    );
+
+    assertEquals(
+      1,
+      emailDatas
+        .stream()
+        .filter(e -> e.recipientName().equals("Foo Bar") && e.recipientAddress().equals("foo@bar"))
+        .count()
+    );
+
     emailDatas.forEach(emailData -> {
-      assertEquals("AKT", emailData.sender());
       assertEquals("Yhteydenotto kääntäjärekisteristä", emailData.subject());
       assertEquals("hello world", emailData.body());
     });
-    assertEquals(1, emailDatas.stream().filter(e -> e.recipient().equals("foo@bar")).count());
   }
 
   @Test
   public void createContactRequestShouldSaveValidRequestWithDuplicateTranslatorIds() {
-    MeetingDate meetingDate = createMeetingDate();
-    initTranslators(meetingDate, 2);
+    final MeetingDate meetingDate = createMeetingDate();
+    final List<Translator> translators = initTranslators(meetingDate, 1);
 
-    final long translatorId = translatorRepository.findAll().get(0).getId();
-    List<Long> translatorIds = List.of(translatorId, translatorId);
+    final List<Translator> duplicateTranslators = List.of(translators.get(0), translators.get(0));
 
-    final ContactRequestDTO contactRequestDTO = createContactRequestDTO(translatorIds, FROM_LANG, TO_LANG);
+    final ContactRequestDTO contactRequestDTO = createContactRequestDTO(duplicateTranslators, FROM_LANG, TO_LANG);
 
-    ContactRequest contactRequest = contactRequestService.createContactRequest(contactRequestDTO);
-    List<ContactRequestTranslator> contactRequestTranslators = contactRequestTranslatorRepository.findAll();
+    final ContactRequest contactRequest = contactRequestService.createContactRequest(contactRequestDTO);
+    final List<ContactRequestTranslator> contactRequestTranslators = contactRequestTranslatorRepository.findAll();
 
     assertEquals(contactRequestDTO.message(), contactRequest.getMessage());
 
     assertEquals(1, contactRequestTranslators.size());
 
-    ContactRequestTranslator ctr = contactRequestTranslators.get(0);
+    final ContactRequestTranslator ctr = contactRequestTranslators.get(0);
 
     assertEquals(contactRequest.getId(), ctr.getContactRequest().getId());
-    assertEquals(translatorId, ctr.getTranslator().getId());
+    assertEquals(translators.get(0).getId(), ctr.getTranslator().getId());
   }
 
   @Test
   public void createContactRequestShouldThrowIllegalArgumentExceptionForNonExistingTranslatorIds() {
-    MeetingDate meetingDate = createMeetingDate();
-    initTranslators(meetingDate, 1);
+    final Translator translator = Factory.translator();
+    translator.setId(0L);
```

Comment:
createContactRequestDTO metodi oikeasti ei tarvitse Translator objektia vaan pelkät id:t, niin onko selkeämpää että olisi toinen createContactRequestDTO joka ottaisi pelkät id:t näitä tapauksia varten kun objektia ei oikeasti ole.